coding-standards-and-naming.rst

```diff
@@ -332,76 +147,47 @@ the symbol names could change in line with these guidelines:
 -  ``openssl_dh_compute_key()`` becomes ``OpenSSL\dh_compute_key()``
 -  ``X509_PURPOSE_SSL_CLIENT`` becomes ``OpenSSL\X509_PURPOSE_SSL_CLIENT``
 
-The above guidelines recommend against the global ``FFI`` class used by the ffi
-extension. Using ``FFI\FFI`` would be preferred.
-
-Core, standard, spl
--------------------
-
-PHP has three extensions that together form the core of the standard library.
-The "Core" extension is part of the Zend Engine, and defines a relatively small
-number of functions and classes. It contains core types like ``stdClass`` and
-``Iterator``, as well as introspection functions like ``get_object_vars()``. The
-"standard" extension contains the majority of the standard library functions,
-including ``array_*()`` and ``str_*()`` functions. The "spl" extension was
-historically the "object-oriented" part of the standard library, containing
-data-structures like ``ArrayObject``, exceptions and iterators.
+The above guidelines recommend against the global ``FFI`` class used by the
+``ffi`` extension. Using ``FFI\FFI`` would be preferred.
 
-The distinction between these three extensions is somewhat murky from an
-end-user perspective, and largely historical. Symbols have moved between these
-extensions, e.g. the ``Iterator`` interface originated in spl, but now lives in
-Core.
+Existing Non-Namespaced Symbols and Consistency
+===============================================
 
-Because these extensions combine a lot of unrelated or only tangentially related
-functionality, symbols should not be namespaced under the ``Core``, ``Standard``
-or ``Spl`` namespaces. Instead, these extensions should be considered as a
-collection of different components, and should be namespaced according to these.
-
-For example, ``str_contains()`` could become ``Str\contains()``, ``fopen()``
-could become ``File\open()``, and ``password_hash()`` could become
-``Password\hash()``. (These are non-normative examples, the RFC does not propose
-using these specific namespaces.)
+When adding new symbols to existing extensions it is RECOMMENDED to be
+consistent with existing symbols, rather than to follow the namespacing
+guidelines.
 
-Existing non-namespaces symbols and consistency
------------------------------------------------
-
-When adding new symbols to existing extensions, it is more important to be
-consistent with existing symbols than to follow the namespacing guidelines.
-
-For example, the ``array_is_list()`` function added in PHP 8.1 should indeed be
-called ``array_is_list()`` and should not be introduced as ``Array\is_list()``
-or similar. Unless and until existing ``array_*()`` functions are aliased under
-an ``Array\*`` namespace, new additions should continue to be of the form
-``array_*()`` to maintain horizontal consistency.
+For example, the ``array_is_list()`` function added in PHP 8.1 MUST indeed be
+called ``array_is_list()`` and MUST NOT have be introduced as
```

Comment:
Is this proper English? have been?

Reply:
yes, `have been` is what I wanted to write.